conans/client/generators/xcode.py

```diff
@@ -26,6 +27,7 @@ def __init__(self, conanfile):
         self.c_compiler_flags = " ".join(deps_cpp_info.cflags)
         self.cpp_compiler_flags = " ".join(deps_cpp_info.cppflags)
         self.linker_flags = " ".join(deps_cpp_info.sharedlinkflags)
+        self.build_paths = " ".join('"%s"' % d.replace("\\", "/") for d in deps_cpp_info.build_paths)
```

Comment:
But the ``build_paths`` are paths to the "build" folders within the package, which are very rarely set. Maybe it should point to the "lib" or "bin" folder? Or to both? 

At least the issue https://github.com/conan-io/conan/issues/3161 is not defining ``builddirs`` at all.

Reply:
By default is pointing to the root, you told me when I was developing the new CMake generators, right?

Reply:
```A framework is a hierarchical directory that encapsulates shared resources, such as a dynamic shared library, nib files, image files, localized strings, header files, and reference documentation in a single package.``` I don't know if they should be in `lib` or `bin` or just the root.

Reply:
Should I replace with roots directly to avoid confusions?

Reply:
Yes, it is pointing at the root, because it is where CMake scripts or other scripts typically are. But frameworks are more like a library. Maybe having it in the root only is a more restrictive, I am not sure what would be the possible layouts for framework. Maybe ask the user @BillHoo?

Reply:
The layout of an apple framework is something like:
```
Headers		Modules
```

The `Modules` can have many other folders inside.

Reply:
As for me, I put the Demo.framework into root, cos' its not only a bin, lib or header of C++, its a bundle of those things, but maybe someothers will treat it like a bin or lib for Mac, who knows. Happy to see the update. 👍